salt/grains/metadata_gce.py

```diff
@@ -15,72 +15,47 @@
 
 """
 
-import logging
-import os
-
-import salt.utils.data
 import salt.utils.http as http
 import salt.utils.json
-import salt.utils.stringutils
-
-# metadata server information
-HOST = "http://169.254.169.254/"
+import logging
 
+__virtualname__ = "gce"
+IP = "169.254.169.254"
+URL = f"http://{IP}/computeMetadata/v1/?alt=json&recursive=true"
 log = logging.getLogger(__name__)
 
-
 def __virtual__():
+    checks = []
+    # Check if metadata_server_grains minion option is enabled
     if __opts__.get("metadata_server_grains", False) is False:
+        checks.append(False)
```

Comment:
Iit should return here immediatelly... if `metadata_server_grains` is not specified and `169.254.169.254` is not reachable (for example on systems where connectivity to this IP would result in connection timed out), minion will slow down considerably, as grains are loaded every time any action is done via minion. My suggestion is to avoid the `checks` variable altogether and `return False` from `__virtual__` as soon as we know something is not ok

Reply:
I;ve committed the suggest change - thanks for pointing out the potential issue.

Reply:
I've pushed another commit - just to reuse the original __virtual__ function. i misunderstood your original point that, that the __opts__ check should bail immediately, but the other checks are valid, just should not be lumped all together.